pool/app/tags/entity.ml

```diff
@@ -1,29 +1,65 @@
-module Id = struct
-  type t = string
+open CCFun.Infix
+module Message = Pool_common.Message
+module Ptime = Pool_common.Model.Ptime
+module Id = Pool_common.Id
+
+let printer = Utils.ppx_printer
+
+module Model = struct
+  module Core = struct
+    let field = Message.Field.Model
+
+    type t =
+      | Contact [@name "contact"] [@printer printer "contact"]
+      | Experiment [@name "experiment"] [@printer printer "experiment"]
+      | Session [@name "session"] [@printer printer "session"]
+    [@@deriving enum, eq, ord, sexp_of, show { with_path = false }, yojson]
+  end
+
+  include Pool_common.Model.SelectorType (Core)
+  include Core
 end
 
 module Title = struct
-  type t = string
-end
+  include Pool_common.Model.String
 
-module Relation = struct
-  type t =
-    | Participant
-    | Experiment
-    | Pool
+  let field = Message.Field.Title
+  let schema () = schema field ()
 end
 
 module Description = struct
-  type t = string
+  include Pool_common.Model.String
+
+  let field = Message.Field.Description
+  let schema () = schema field ()
 end
 
-type 'a tag =
+type t =
   { id : Id.t
   ; title : Title.t
-  ; description : Description.t
-  ; relation : Relation.t
-  ; owner : Admin.t
-  ; created_at : Ptime.t
-  ; updated_at : Ptime.t
+  ; description : Description.t option
   }
 [@@deriving eq, show]
+
+let create ?(id = Id.create ()) ?description title =
+  let open CCResult in
+  let* description =
+    CCOption.map_or
+      ~default:(Ok None)
+      (Description.create %> map CCOption.return)
+      description
+  in
+  let* title = Title.create title in
+  Ok { id; title; description }
+;;
+
+module Tagged = struct
+  type t =
+    { model : Model.t
```

Comment:
That information should be stored in the Tags.t type to differentiate the tags available for different models. 

As we use UUIDs, we do not need this information here.

Reply:
moved model information to tag itself